### ISSUE TITLE
syntax->string doesn't work on syntax pairs

### ISSUE THREAD

**Issue by endobson** [racket/racket#1629]
```
(syntax->string #'((a . b)))
; map: contract violation
;   expected: list?
;   given: #f
;   argument position: 2nd
; [,bt for context]
```

Bug is here where it assumes that if the syntax object is a syntax pair then it is a syntax list.
https://github.com/racket/racket/blob/master/racket/collects/syntax/to-string.rkt#L61

**Comment by lkh01** [comment]
Hey, I came across this issue [here](https://github.com/racket/racket/wiki/Easy-bugs-to-fix) and I want to fix it and https://github.com/racket/racket/issues/1628.

It looks like it's still broken in the current build (v7.5.0.7), so I'll take a shot at it.

**Comment by samth** [comment]
Fixed by #2911
